### PR TITLE
#5 unicode decode aw

### LINKED ISSUE #5: UnicodeDecodeError when training synthetics on Kaggle dataset
Running on a downloaded Kaggle dataset: /home/amy/sdata/US_household_income_NH.csv (my cloud server), I get the error:

~/env/ml/lib/python3.7/codecs.py in decode(self, input, final)
    320         # decode input (taking the buffer into account)
    321         data = self.buffer + input
--> 322         (result, consumed) = self._buffer_decode(data, self.errors, final)
    323         # keep undecoded input until the next call
    324         self.buffer = data[consumed:]

UnicodeDecodeError: 'utf-8' codec can't decode byte 0xf1 in position 4016: invalid continuation byte

Running iconv -c -f utf-8 -t ascii on the file resolves the error

### PR DESCRIPTION
**Pull request by zredlined**
Fixes #5 
* Support CRLF newlines in training datasets

Updates:
* Specify max_lines in configuration file vs. max_chars (more intuitive)
* Commas & newlines treated by tokenizer as user defined symbols (decreases records in test data failing validation by 60%)
* Increased default vocabulary size from 200 to 15000. Increases number of successful record validations in tests.

### CODE REVIEW

File: src/gretel_synthetics/train.py [hunk 1]
```diff
@@ -59,9 +60,13 @@ def annotate_training_data(store: BaseConfig):
     # required for sentencepiece to tokenize newline characters
     logging.info(f"Annotating training data from {store.input_data}")
     training_text = []
-    with open(store.input_data, 'r', encoding='utf-8') as infile:
+    with open(store.input_data, 'r', encoding='utf-8', errors='replace') as infile:
         for line in infile:
-            training_text.append(f"{line.strip()}")
+            if (store.max_lines == 0) or (len(training_text) < store.max_lines):
```
Comment (misberner) [inline src/gretel_synthetics/train.py:65]: nit: I prefer moving the `break` case up so as to minimize indentation of the happy-path code for better readability, i.e.,
```
if store.max_lines and len(training_text) >= store.max_lines:
  break
line = line.strip()....
```
(feel free to ignore)

File: src/gretel_synthetics/train.py [hunk 2]
```diff
@@ -59,9 +60,13 @@ def annotate_training_data(store: BaseConfig):
...
+                line = line.strip().replace(",", "<c>")
+                training_text.append(f"{line}")
```
Comment (misberner) [inline src/gretel_synthetics/train.py:67]: This can just be `line`, no need for an `f"..."` literal

File: examples/synthetic_records.ipynb [hunk 3]
```diff
@@ -41,12 +41,12 @@
     "from gretel_synthetics.config import LocalConfig\n",
     "\n",
     "# Create a config that we can use for both training and generating, with CPU-friendly settings\n",
-    "# The default values for ``max_chars`` and ``epochs`` are better suited for GPUs\n",
+    "# The default values for ``max_lines`` and ``epochs`` are better suited for GPUs\n",
     "\n",
     "config = LocalConfig(\n",
-    "    max_chars=100000,  # friendly towards CPUs\n",
+    "    max_lines=0,  # friendly towards CPUs\n",
```
Comment (misberner) [inline examples/synthetic_records.ipynb:47]: There's a mismatch here. This is also the default value, which per the above comment is better suited for GPUs and hence not CPU friendly. This should thus be a value other than `0`, or entirely removed, along with its mention in the comment 3 lines above.

File: src/gretel_synthetics/train.py (hunk 1 above)
Comment (zredlined) [reply src/gretel_synthetics/train.py:65]: that's definitely more clean- updated

File: examples/synthetic_records.ipynb (hunk 3 above)
Comment (zredlined) [reply examples/synthetic_records.ipynb:47]: nice catch- updated to a sane number for training on CPU (5000 lines)

**Pull request merged**